packages/workspace/src/browser/workspace-service.ts

```diff
@@ -32,10 +40,14 @@ import { WorkspacePreferences } from './workspace-preferences';
 export class WorkspaceService implements FrontendApplicationContribution {
 
     private _root: FileStat | undefined;
```

Comment:
I like the idea of having `get workspaceDescription()` instead of  `get root()` & `get workspaceRootChange()`. And it was the way I implemented it in the first place. But ...

Correct me if I am wrong:
The value of `workspaceDescription` is dependent on the readiness of `this.preferences` and the value of `this.preferences['workspace.supportMultiRootWorkspace']`, while the readiness of `this.preferences` is counting on the value of `workspaceService.root` (`preferences` requires the value of `root` to access the settings.json file). 

This is why I don't think we are able to get rid of the `get root()` before the `preference` extension is made multi-root compatible. So, what you are suggesting is in my TODO list, but it wouldn't be done in this PR. Are you OK with it ?


(The 1st and 2nd bullet points in your comment can be done. My words above are only for the 3rd and 4th bullet points.)

Reply:
ok

Reply:
I agree with Anton. We basically have to informations that need to be kept in fields:
 1) `workspaceDescription` (a FileStat pointing to either a file or a description file)
 2) `workspaceRoots`(array of FileStat pointing to the workspace roots)